projects/etcd-io/etcd/docker/linux/Dockerfile

```diff
@@ -13,11 +13,18 @@
 # limitations under the License.
 
 ARG BASE_IMAGE
-FROM $BASE_IMAGE
+FROM $BASE_IMAGE AS base
 
 ARG TARGETARCH
 ARG TARGETOS
 
+FROM base AS branch-version-amd64
+
+FROM base AS branch-version-arm64
+ENV ETCD_UNSUPPORTED_ARCH=arm64
+
+FROM branch-version-${TARGETARCH} AS final
```

Comment:
Is `final` special here?  Im just now sure how the process running the build knows to that tag that as the actual image?

Reply:
I don't think final is a key word or anything. I look at that as more of a comment to let you know that is the final stage of a multistage build.

As I understand it, arm64 and amd64 both get tagged the same and when you pull, it attempts to pull the best match for your architecture. When I'm on my mac I run a `docker pull` of one of those images and I get an amd64 version. On the Pi I the same command gives me the arm64 version.

Reply:
ok that makes sense